src/frontend/mame/ui/icorender.cpp

```diff
@@ -179,7 +179,7 @@ bool load_ico_image(util::core_file &fp, unsigned count, unsigned index, bitmap_
 {
 	// read the directory entry
 	std::error_condition err;
-	size_t actual;
+	size_t actual = 0;
```

Comment:
The trouble with this is it will mask other issues when running under a memory analyser.  The variable can’t be used uninitialised:
* If `err` is clear at line 185, `fp.read` will set `actual`.
* `actual` will only be checked at line 187 if `err` is clear – you can’t get here with `err` clear if `fp.read` has not been set.

If you initialise the variable like this, then it can mask bugs where a code path allows `actual` to be used without being set because the zero initialisation counts as being set for the purpose of the memory analyser.

Reply:
MSVC has lots of issues with spurious C2220 – is it not possible to configure it to not treat C2220 as an error?  As far as I know our default settings for MSVC projects don’t treat warnings as errors at all because of spurious errors like this.

Reply:
I'm using the project files generated using `make vs2019` as per the documentation, maybe the project files generated contain the /WX parameter somewhere? If so, that's another bug.